### ISSUE TITLE
Automating alpha

### ISSUE THREAD

**Issue by lgvaz** [rail-berkeley/softlearning#17]
Alpha loss is defined as ([code](https://github.com/rail-berkeley/softlearning/blob/master/softlearning/algorithms/sac.py#L283)):  
`alpha_loss = -tf.reduce_mean(log_alpha * tf.stop_gradient(log_pis + self._target_entropy))`  

In other words:
`alpha_loss = log_alpha * (- <Negative constant>)`  
`alpha_loss = log_alpha * (<Positive constant>)`

So minimizing `alpha_loss` means minimizing `log_alpha`, this means that `alpha`always goes to zero no matter what, and this is indeed what I'm confirming in my experiments.  

I'm obviously forgeting something, but I'm not being able to figure it out.

**Comment by hartikainen** [comment]
Good question. I think the confusion here comes from the range of the `tf.stop_gradient(log_pis + self._target_entropy)` term, which can in fact be either positive or negative due to `log_pis` being the log density of the probability of action a at state s. @haarnoja correct me if I'm wrong.

**Comment by haarnoja** [comment]
yeah, the magnitude of `log_pis + self._target_entropy` can be either positive or negative. The gradient vanishes when `log_pis == self._target_entropy` as desired.
